### PR TITLE
WIP DO NOT MERGE: FIXES RHOARDOC-1144:initial logging changes

### PR DESCRIPTION
**Pull request by lavernw1**
@rhoads-zach @inoxx03 @tradej  initial content for logging - couple updates needed before sme review..

### CODE REVIEW

File: docs/topics/proc_add-debug-logging-spring-boot.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+= Add Debug Logging
+There are several options for debug logging. This example uses the JAVA-OPTIONS environment variable.
```
Comment (Ladicek) [inline docs/topics/proc_add-debug-logging-spring-boot.adoc]: It's not `JAVA-OPTIONS`, it's `JAVA_OPTIONS`.

File: docs/topics/proc_add-debug-logging-spring-boot.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+. Add debug statements.
++
+[source,js,options="nowrap",subs="attributes+"]
+----
+app.use('/api/greeting', (request, response) => {
+  const name = request.query ? request.query.name : undefined;
+  //log name in debugging
+  debug('name: '+name);
+  response.send({content: `Hello, ${name || 'World'}`});
+});
```
Comment (Ladicek) [inline docs/topics/proc_add-debug-logging-spring-boot.adoc]: Is this really Spring Boot? :-)

File: docs/topics/proc_show-debug-logging-local-spring-boot.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+.Prerequisites
+* An application with debug logging.
+
+.Procedure
+.  Deploy your application with fabric8:debug to enable debug logging.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ mvn clean fabric8:debug
```
Comment (Ladicek) [inline docs/topics/proc_show-debug-logging-local-spring-boot.adoc]: This doesn't enable debug logging. This enables connecting a remote debugger. (Which is interesting, I didn't know about this possibility. CC @tradej)

File: docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+. Return to your pod logs and notice there are no debug logging messages in the logs.
+
+. Set the `DEBUG` environment variable to enable debug logging.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ oc get dc
+...
+$ oc set env dc DC_NAME DEBUG=mybooster
```
Comment (Ladicek) [inline docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc]: This is just specific to the Node.js debug logging technique and doesn't really translate to Spring Boot directly.

**Comment by Ladicek** [comment]
Seems like copy&paste from Node.js, which is way too different from Java-based runtimes.

**Comment by lavernw1** [comment]
@Ladicek  - I havent added the Java relate updates for Spring boot. This is an outline. My apologies it was not ready for your review.

**Comment by Ladicek** [comment]
Ah OK! Sorry about the noise then :-)

**Comment by lavernw1** [comment]
@rhoads-zach @inoxx03 @tradej @Ladicek  Updated WIP PR and Provided updates for logging. Take a look.

File: docs/topics/proc_add-debug-logging-spring-boot.adoc [hunk 5]
```diff
@@ -0,0 +1,113 @@
+= Add Debug Logging
+There are several options for debug logging. This example uses the Arquillian Testrunner JUnit Container.
```
Comment (Ladicek) [inline docs/topics/proc_add-debug-logging-spring-boot.adoc]: Arquillian is totally unrelated, and this entire file doesn't really say anything about logging.

File: docs/topics/proc_add-tomcat-logging.adoc [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+ This example adds configuration necessary to send the Embedded Tomcat containers’ access logging to stdout.
+
+
+.Prerequisites
+* An application you want to enable Tomcat logging for. For example, xref:mission-http-api-spring-boot[a booster].
+*
+
+.Procedure
+
+.  Set `server.use-forward-headers` to  `true` in your application.yml file:
```
Comment (Ladicek) [inline docs/topics/proc_add-tomcat-logging.adoc]: This is totally unrelated to logging.

File: docs/topics/proc_add-tomcat-logging.adoc [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+. Specify logging levels (e.g. DEBUG) for the  ` org.apache.tomcat` package and specify the name of the logging file.
++
+[source,js,options="nowrap",subs="attributes+"]
+----
+logging:
+  level.org.apache.tomcat: "DEBUG"
+  file: "application.log"
+----
+
+. Configure your embedded Tomcat container and add `logback-access` by defining a provider method.
```
Comment (Ladicek) [inline docs/topics/proc_add-tomcat-logging.adoc]: This should read something like "Add a `LogbackValve` to your embedded Tomcat container to use the Logback logging framework."

File: docs/topics/proc_show-debug-logging-local-spring-boot.adoc [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+.Prerequisites
+* An application with debug logging.
+
+.Procedure
+.  Deploy your application to enable debug logging.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ mvn clean fabric8:debug
```
Comment (Ladicek) [inline docs/topics/proc_show-debug-logging-local-spring-boot.adoc]: So it this on localhost? If you're doing `fabric8:debug`, it's on OpenShift, right?

File: docs/topics/proc_show-debug-logging-local-spring-boot.adoc [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ curl http://localhost:8080/api/greeting?name=Sarah
+----
+
+. View your application logs to see your debug messages.
++
+[source,options="nowrap",subs="attributes+"]
+----
+myapp name: Sarah +3m
```
Comment (Ladicek) [inline docs/topics/proc_show-debug-logging-local-spring-boot.adoc]: This logging message is copied&pasted from the Node.js guide. In Spring Boot, the log message will most likely look very different.

File: docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+. Return to your pod logs and notice there are no debug logging messages in the logs.
+
+. Set the `JAVA_DEBUG` environment variable to enable debug logging.
++
+[source,bash,options="nowrap",subs="attributes+"]
+----
+$ oc get dc
+...
+$ oc set env dc DC_NAME JAVA_DEBUG=mybooster
```
Comment (Ladicek) [inline docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc]: What's this `JAVA_DEBUG=mybooster`? It should be `JAVA_DEBUG=true`.

File: docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+----
+
+. Return to your pod logs to watch the update roll out.
```
Comment (Ladicek) [inline docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc]: The pod will shut down and a new one will be created. This won't work.

File: docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+...
+$ curl $APPLICATION_ROUTE/api/greeting?name=Sarah
+----
+
+. View the application logs to see the debug messages.
++
+[source,options="nowrap",subs="attributes+"]
+----
+...
+mybooster name: Sarah +3m
```
Comment (Ladicek) [inline docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc]: Again, this message is copied&pasted from Node.js guide, with Spring Boot, it will be very different.

**Comment by Ladicek** [comment]
This needs a very thorough SME review really.

File: docs/topics/proc_add-debug-logging-spring-boot.adoc (hunk 5 above)
Comment (lavernw1) [reply docs/topics/proc_add-debug-logging-spring-boot.adoc]: @Ladicek This was just used to add debugging statements for the app. Can update the text.  Yes I will get a SME review as well thanks.

**Comment by lavernw1** [comment]
@Ladicek thanks for your comments, will have SME review.

File: docs/topics/proc_add-tomcat-logging.adoc (hunk 7 above)
Comment (lavernw1) [reply docs/topics/proc_add-tomcat-logging.adoc]: @Ladicek will update  comment per your suggestion.

File: docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc (hunk 12 above)
Comment (lavernw1) [reply docs/topics/proc_show-debug-logging-openshift-spring-boot.adoc]: @Ladicek thanks for the comments will have a SME review and update expected output.

File: docs/topics/proc_show-debug-logging-local-spring-boot.adoc (hunk 8 above)
Comment (lavernw1) [reply docs/topics/proc_show-debug-logging-local-spring-boot.adoc]: @Ladicek youre right, I was thinking OpenShift/Minishift  for this content. But this should be an example on your local host/pc (non minishift), so will change this example.

**Comment by lavernw1** [comment]
@Ladicek  @rhoads-zach @inoxx03 @tradej @metacosm @cmoulliard   These were interim updates not meant for review. Please hold your review until the entire PR has been updated. Will give  you an update when its ready for review. Thank You
